src/main/java/org/terasology/gookeeper/system/VisitorSystem.java

```diff
@@ -272,21 +287,24 @@ public void onPeriodicAction(PeriodicActionTriggeredEvent event, EntityRef entit
             Quat4f rotation = Quat4f.shortestArcQuat(Direction.FORWARD.getVector3f(), dir);
 
             if (visitorPrefab.isPresent() && visitorPrefab.get().getComponent(LocationComponent.class) != null) {
-                entityManager.create(visitorPrefab.get(), spawnPos, rotation);
+                EntityRef visitor = entityManager.create(visitorPrefab.get(), spawnPos, rotation);
+                VisitorComponent visitorComponent = visitor.getComponent(VisitorComponent.class);
+                visitorComponent.visitorEntranceBlock = entityRef;
+                visitor.saveComponent(visitorComponent);
             }
         }
     }
 
     /**
      * Receives ActivateEvent when the targeted pen block is activated, and prints the ID of the corresponding pen.
      *
-     * @param event,entity   The ActivateEvent, the instigator entity
+     * @param event,entity   The ActivateEvent
```

Comment:
Weird whitespacing here. Each arg really should be on its own line (`/** <enter>` should auto-generate a full template block if needed)